components/MapView.js

```diff
@@ -446,8 +446,45 @@ class MapView extends React.Component {
   }
 
   takeSnapshot(width, height, region, callback) {
-    const finalRegion = region || this.props.region || this.props.initialRegion;
-    this._runCommand('takeSnapshot', [width, height, finalRegion, callback]);
+    return new Promise((resolve, reject) => {
+      let options;
+      if (typeof width === 'object') {
+        options = width;
+      } else {
+        options = {
```

Comment:
Done. I've updated the function so that it uses the old API when 4 arguments are specified (and only when on iOS).  In all other cases the new API is used.
The deprecation message is generated using `console.warn`.